crates/cxx-qt-lib/src/core/qbytearray.cpp

```diff
@@ -161,5 +161,29 @@ qbytearrayResize(QByteArray& byteArray, ::rust::isize size)
 #endif
 }
 
+QByteArray
+qbytearraySimplified(const QByteArray& byteArray)
```

Comment:
Why are these needed? Can't this just be declared directly with CXX?

Reply:
See the comment above them in the header
```
// If Q_COMPILER_REF_QUALIFIERS is set the definition of these is
// T method() const& which CXX doesn't bind it.
```

Basically they are defined as `QByteArray QByteArray::simplified() const&` i wouldn't find a way to bind to a `const&` method in CXX as others are just `const`

Reply:
But i could be missing something obvious :-)

Reply:
```cpp
#if !defined(Q_CLANG_QDOC)
    [[nodiscard]] QByteArraytoLower() const &
    { return toLower_helper(*this); }
    [[nodiscard]] QByteArray toLower() &&
    { return toLower_helper(*this); }
    [[nodiscard]] QByteArray toUpper() const &
    { return toUpper_helper(*this); }
    [[nodiscard]] QByteArray toUpper() &&
    { return toUpper_helper(*this); }
    // ...
#else
    [[nodiscard]] QByteArray toLower() const;
    [[nodiscard]] QByteArray toUpper() const;
    [[nodiscard]] QByteArray trimmed() const;
    [[nodiscard]] QByteArray simplified() const;
#endif
```
Ooooof, it so they're disabling the `const&` feature just for the docs, which is why I didn't see that when browsing the Qt documentation...

This is probably done for optimization purposes I'd guess :thinking: 

But then it makes sense that we need this.
Something that upstream would need to address, but probably not worth the effort.

Reply:
Nothing we can fix right now.